urllib3/poolmanager.py

```diff
@@ -42,12 +42,39 @@
     'HTTPSPoolKey', HTTPPoolKey._fields + SSL_KEYWORDS
 )
 
-# The default pool key for each scheme; each PoolManager makes a copy of this
-# dictionary so they can be configured globally here, or individually on the
-# instance.
-pool_keys_by_scheme = {
-    'http': HTTPPoolKey,
-    'https': HTTPSPoolKey,
+
+def default_key_normalizer(request_context, key_class):
+    """
+    Create a pool key of type ``key_class`` for a request.
+
+    According to RFC 3986, both the scheme and host are case-insensitive.
+    Therefore, this function normalizes both before constructing the pool
+    key for an HTTPS request. If you wish to change this behaviour, provide
+    alternate callables to ``pool_key_funcs_by_scheme``.
+
+    :param request_context:
+        A dictionary-like object that contain the context for a request.
+        It should contain a key for each field in the :class:`HTTPPoolKey`
+
+    :param key_class:
+        The class to use when constructing the key. This should be a namedtuple
+        with the ``scheme`` and ``host`` keys at a minimum.
+    """
+    context = {}
+    for key in key_class._fields:
+        context[key] = request_context.get(key)
+    context['scheme'] = context['scheme'].lower()
+    context['host'] = context['host'].lower()
+    return key_class(**context)
+
+
+# A dictionary that maps a scheme to a callable that creates a pool key.
+# This can be used to alter the way pool keys are constructed, if desired.
+# Each PoolManager makes a copy of this dictionary so they can be configured
+# globally here, or individually on the instance.
+pool_key_funcs_by_scheme = {
```

Comment:
IMO `key_func_by_scheme` (or `key_fn_by_scheme`?)

Honestly could really use a fresher idea for namespacing these registries, the current naming scheme is getting out of hand.